homeassistant/components/compensation/sensor.py

```diff
@@ -0,0 +1,144 @@
+"""Support for compensation sensor."""
+import logging
+
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    ATTR_UNIT_OF_MEASUREMENT,
+    CONF_ATTRIBUTE,
+    CONF_ENTITY_ID,
+    CONF_NAME,
+    CONF_UNIT_OF_MEASUREMENT,
+    STATE_UNKNOWN,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_state_change_event
+
+from .const import CONF_COMPENSATION, CONF_POLYNOMIAL, CONF_PRECISION, DATA_COMPENSATION
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_ATTRIBUTE = "attribute"
+ATTR_COEFFICIENTS = "coefficients"
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the Compensation sensor."""
+    if discovery_info is None:
+        return
+
+    compensation = discovery_info.get(CONF_COMPENSATION)
+    conf = hass.data[DATA_COMPENSATION][compensation]
+
+    async_add_entities(
+        [
+            CompensationSensor(
+                hass,
+                conf[CONF_ENTITY_ID],
+                conf.get(CONF_NAME),
+                conf.get(CONF_ATTRIBUTE),
+                conf[CONF_PRECISION],
+                conf[CONF_POLYNOMIAL],
+                conf.get(CONF_UNIT_OF_MEASUREMENT),
+            )
+        ]
+    )
+
+
+class CompensationSensor(Entity):
```

Comment:
(needs a rebase for this one)

```suggestion
class CompensationSensor(SensorEntity):
```